src/backend/PortabilityService.ConfigurationSource/ConfigurationSource.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Extensions.Configuration.Memory;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace PortabilityService.ConfigurationSource
+{
+    public class PortabilityServiceConfigurationSource : MemoryConfigurationSource
+    {
+        private const string DefaultConfigurationServiceUrl = "http://localhost:1443";
+        internal const string PortabilityServiceConfigurationRoot = "PortabilityServiceSettings";
+
+        public PortabilityServiceConfigurationSource(string configurationSection = PortabilityServiceConfigurationRoot)
+            : this(GetConfigurationServiceUrlFromEnvironment() ?? new Uri(DefaultConfigurationServiceUrl), configurationSection) { }
+
+        public PortabilityServiceConfigurationSource(Uri configurationServiceUrl, string configurationSection = PortabilityServiceConfigurationRoot)
+            : this(CreateHttpClient(configurationServiceUrl ?? throw new ArgumentNullException(nameof(configurationServiceUrl))), configurationSection ?? throw new ArgumentNullException(nameof(configurationSection))) { }
+
+        public PortabilityServiceConfigurationSource(HttpClient httpClient, string configurationSection = PortabilityServiceConfigurationRoot)
+        {
+            InitialData = GetConfigurationSectionSettings(configurationSection ?? throw new ArgumentNullException(nameof(configurationSection)), httpClient ?? throw new ArgumentNullException(nameof(httpClient))).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Gets the Portability configuration service URL from the environment
+        /// </summary>
+        /// <returns></returns>
+        private static Uri GetConfigurationServiceUrlFromEnvironment()
+        {
+            var urlFromEnvironment = Environment.GetEnvironmentVariable("PortabilityConfigurationServiceUrl");
```

Comment:
I would put this in a constants class above.

Reply:
Added to constants class